### PR TITLE
Add Playtester feature

### PR DESCRIPTION
**Pull request by sr229**
This allows the mod to be publicly listed but in the following conditions:

-  The author has to set a Sayonika-registered user to access downloads.

Should the requesting UA is authenticated and ID is enrolled to the private beta, we allow downloads, if not, we restrict them, regardless of authenticated request.

### CODE REVIEW

File: routes/mods.py [hunk 1]
```diff
@@ -129,9 +130,25 @@ def dict_all(models):
 
         authors.append({"id": user_id, "role": AuthorRole.Owner})
 
+        if is_private_beta is not None:
+            mod.is_private_beta = is_private_beta
+
+        if playtesters is not None:
+            for playtester in playtesters:
```
Comment (Ovyerus) [inline routes/mods.py]: Could probably just ditch the `unique_id` list here and use `seen` in future, as it'd be unique, or ditch the seen set and use `unique_id`. Not really any need for 2 things.

File: routes/mods.py [hunk 2]
```diff
@@ -169,16 +186,19 @@ def dict_all(models):
...
         "icon": None
     }, locations=("json",))
-    async def patch_mod(self, mod_id: str = None, **kwargs):
+    async def patch_mod(self, mod_id: str = None, is_private_beta: bool = None, playtesters: List[str] = None, **kwargs):
         if not await Mod.exists(mod_id):
             abort(404, "Unknown mod")
 
         mod = await Mod.get(mod_id)
         updates = mod.update()
+        playtesters = await Playtesters.get(mod_id)
```
Comment (Ovyerus) [inline routes/mods.py]: Rename the variable so it doesn't conflict with the argument, and this function won't work how you want. Try:
```py
await Playtesters.query.where(Playtesters.mod_id = mod_id).gino.all()
```

File: framework/models/mod.py [hunk 3]
```diff
@@ -16,6 +16,7 @@ class Mod(db.Model, Base):
     tagline = db.Column(db.Unicode(100))
     description = db.Column(db.Unicode(10000))
     website = db.Column(db.Unicode())
+    is_private_beta = db.Column(db.Boolean())
```
Comment (Ovyerus) [inline framework/models/mod.py]: Add `default=False`, in case people don't specify.

File: routes/mods.py [hunk 4]
```diff
@@ -136,8 +136,6 @@ def dict_all(models):
         if playtesters is not None:
             for playtester in playtesters:
                 seen = set()
-                unique_id = []
```
Comment (Ovyerus) [inline routes/mods.py]: Still hasn't been changed in other places.

Comment (sr229) [reply routes/mods.py]: I removed that though in accordance to what you told me

**Pull request merged**